slahmr/run_rerun_vis.py

```diff
@@ -0,0 +1,331 @@
+            if len(good_joints_xy):
+                rr.log_line_segments(
+                    f"world/{phase_label}/camera/image/skeleton/#{i}",
+                    good_joints_xy.reshape(-2, 2),
+                    color=_index_to_color(i),
+                )
+            else:
+                rr.log_cleared(f"world/{phase_label}/camera/image/skeleton/#{i}")
+    rr.set_time_sequence(f"frame_id_{phase_label}", None)
+
+
+def log_to_rrd(
+    log_dir: str,
+    dev_id: int,
+    phases: List[str],
+    save_dir=None,
+    phase_labels: Optional[List[str]] = None,
+):
+    print(log_dir)
+    cfg = load_config_from_log(log_dir)
+
+    # make sure we get all necessary inputs
+    cfg.data.sources = expand_source_paths(cfg.data.sources)
+    print("SOURCES", cfg.data.sources)
+    dataset = get_dataset_from_cfg(cfg)
+
+    log_to_rerun(
+        cfg,
+        dataset,
+        log_dir,
+        dev_id,
+        phases=phases,
+        phase_labels=phase_labels,
+        spawn=False,
+    )
+    rr.save(os.path.join(save_dir, "log.rrd"))
+
+
+def launch_rerun_vis(i, args):
+    log_dir = args.log_dirs[i]
+    dev_id = args.gpus[i % len(args.gpus)]
+    os.environ["EGL_DEVICE_ID"] = str(dev_id)
+    os.environ["PYOPENGL_PLATFORM"] = "egl"
+    path_name = log_dir.split(args.log_root)[-1].strip("/")
+    exp_name = "-".join(path_name.split("/")[:2])
+
+    if args.save_root is not None:
+        save_dir = f"{args.save_root}/{exp_name}"
+        os.makedirs(save_dir, exist_ok=True)
+
+        log_to_rrd(
+            log_dir,
+            dev_id,
+            phases=args.phases,
+            save_dir=save_dir,
+            phase_labels=args.phase_labels,
+        )
+    else:
+        print(log_dir)
+        cfg = load_config_from_log(log_dir)
+
+        # make sure we get all necessary inputs
+        cfg.data.sources = expand_source_paths(cfg.data.sources)
+        print("SOURCES", cfg.data.sources)
+        dataset = get_dataset_from_cfg(cfg)
+        log_to_rerun(
+            cfg,
+            dataset,
+            log_dir,
+            dev_id,
+            phases=args.phases,
+            phase_labels=args.phase_labels,
+            spawn=True,
+        )
+
+def main(args):
+    """
+    visualize all runs in root
+    """
+    OmegaConf.register_new_resolver("eval", eval)
+    log_dirs = []
+    for root, subd, files in os.walk(args.log_root):
+        if ".hydra" in subd:
+            log_dirs.append(root)
+    args.log_dirs = log_dirs
+    print(f"FOUND {len(args.log_dirs)} TO RENDER")
+
+    if len(args.gpus) > 1:
+        from torch.multiprocessing import Pool
+
+        torch.multiprocessing.set_start_method("spawn")
+
+        with Pool(processes=len(args.gpus)) as pool:
+            res = pool.starmap(
+                launch_rerun_vis, [(i, args) for i in range(len(args.log_dirs))]
+            )
+        return
+
+    for i in range(len(args.log_dirs)):
+        launch_rerun_vis(i, args)
```

Comment:
I think there are a couple to many layers to the whole "launching rerun" stuff. I may be wrong but I believe it should be possible to simplify it quite some.

Reply:
Changed it to just two functions `main` which parses the arguments and prepares the dataset and `log_to_rerun`.